cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/stream/FileStreamAdmin.java

```diff
@@ -288,58 +305,52 @@ public StreamConfig call() throws IOException {
 
   @Override
   public StreamProperties getProperties(Id.Stream streamId) throws Exception {
+    // User should have read access on the stream to read its properties
+    ensureAccess(streamId.toEntityId(), Action.READ);
     StreamConfig config = getConfig(streamId);
     StreamSpecification spec = streamMetaStore.getStream(streamId);
     return new StreamProperties(config.getTTL(), config.getFormat(), config.getNotificationThresholdMB(),
                                 spec.getDescription());
   }
 
   @Override
-  public void updateConfig(final Id.Stream streamId, final StreamProperties properties) throws IOException {
+  public void updateConfig(final Id.Stream streamId, final StreamProperties properties) throws Exception {
     Location streamLocation;
-    try {
-      streamLocation = impersonator.doAs(new NamespaceId(streamId.getNamespaceId()), new Callable<Location>() {
-        @Override
-        public Location call() throws Exception {
-          return getStreamLocation(streamId);
-        }
-      });
-    } catch (Exception e) {
-      Throwables.propagateIfPossible(e, IOException.class);
```

Comment:
I just removed the try/catch block. Diff doesn't show that well. The try/catch block was not required since I changed the method signature to throw Exception instead of IOException.